pkg/cluster/cluster.go

```diff
@@ -793,11 +814,32 @@ func (c *cluster) updateMembers() error {
 		return err
 	}
 
-	c.members.updateClusterMembers(resp.Members)
-
+	if c.members == nil {
+		c.UpdatePeerURLs(resp.Members)
+	} else {
+		c.members.updateClusterMembers(resp.Members)
+	}
 	return nil
 }
 
+// UpdatePeerURLs updates peerURLs according the cluster role.
+func (c *cluster) UpdatePeerURLs(pbMembers []*pb.Member) {
+	primaryMembers := pbMembersToMembersSlice(pbMembers)
+	if c.opt.ClusterRole == "secondary" {
+		newPeerURLs := make([]string, 0)
+		for _, member := range primaryMembers {
+			newPeerURLs = append(newPeerURLs, member.PeerURL)
+		}
+		c.opt.SetClusterPrimaryListenPeerURLs(newPeerURLs)
```

Comment:
I don't think it's a good idea to store back to option, which should honestly react to the input from the command line or config file.

Reply:
I do agree that the options is not the best place to store the cluster node's PeerURLs. I did this, to support adding and removing the primary members, as the set of primary members (and their URLs) change.

Another way to support adding new primary members is described [here](https://github.com/megaease/easegress/issues/233#issuecomment-977359212) in case 3. So basically instead of members listening for cluster modifications, user should explicitly call `egctl member add NAME --peer-urls=NEW-HOSTS` and same for removal. Then this `egctl` call could modify the _options_ of each member. I think this the way etcd advertise in their [tutorial](https://etcd.io/docs/v3.5/tutorials/how-to-deal-with-membership/).